### PR TITLE
[Model Monitoring] Organize the monitoring deployment module

### PR DESCRIPTION
**Pull request by jond01**
Yet another mini-PR related to [ML-5865](https://iguazio.atlassian.net/browse/ML-5865).

Move common parameters to the `MonitoringDeployment` object, instead of passing them from method to method.

Note: this PR should be merged after #5296.

[ML-5865]: https://iguazio.atlassian.net/browse/ML-5865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/api/api/endpoints/functions.py [hunk 1]
```diff
@@ -798,14 +798,16 @@ def _deploy_nuclio_runtime(
             )
         if monitoring_application:
             monitoring_deploy = (
-                server.api.crud.model_monitoring.deployment.MonitoringDeployment()
+                server.api.crud.model_monitoring.deployment.MonitoringDeployment(
+                    project=fn.metadata.project,
+                    auth_info=auth_info,
+                    model_monitoring_access_key=model_monitoring_access_key,
+                    db_session=db_session,
```
Comment (gtopper) [inline server/api/api/endpoints/functions.py]: Let's keep the same parameter order every time.

File: server/api/crud/model_monitoring/deployment.py [hunk 2]
```diff
@@ -422,10 +341,7 @@ def deploy_model_monitoring_writer_application(
 
     def _initial_model_monitoring_stream_processing_function(
         self,
-        project: str,
-        model_monitoring_access_key: str,
         stream_image: str,
-        auth_info: mlrun.common.schemas.AuthInfo,
```
Comment (gtopper) [inline server/api/crud/model_monitoring/deployment.py:428]: Need to remove them from docstring as well.

Comment (jond01) [reply server/api/crud/model_monitoring/deployment.py:428]: Good catch, I think that was the only place. Done in f3e87a4.

File: server/api/api/endpoints/functions.py (hunk 1 above)
Comment (jond01) [reply server/api/api/endpoints/functions.py]: Done in 441f0a5.

**Review by davesh0812 — APPROVED** [review]

**Pull request merged**